### PR TITLE
Add eslint and fix all violations

### PR DESCRIPTION
**Pull request by dignite**
Add eslint to project and apply pluralsight eslint rules. These were found in commit c7d5860

There are many violations but by adding the linting they are shown. We can then solve the rules one by one, for example using [`npx eslint-nibble .`](https://github.com/IanVS/eslint-nibble) and then merge this PR.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
+{
+  "extends": [
+    "eslint:recommended",
+    "plugin:node/recommended",
+    "plugin:promise/recommended",
```
Comment (allan-stewart) [inline .eslintrc.json]: I'm not sure we use any promise code or async/await... should we keep this in anyway?

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    "generator-star-spacing": "off",
+    "mocha/no-exclusive-tests": "error",
+    "node/no-unpublished-require": [
+      "error",
+      {
+        "allowModules": [
+          "electron",
+          "chai",
+          "sinon",
+          "sinon-chai"
```
Comment (allan-stewart) [inline .eslintrc.json]: We aren't using chai yet; should we remove these until (if?) we introduce chai?

File: .eslintrc.json (hunk 1 above)
Comment (dignite) [reply .eslintrc.json]: We need to weigh the cost of having it therr now vs the risk that we forget to add it when it is needed. I'm fine either way but leaning towards *keep*.

**Comment by allan-stewart** [comment]
Thanks for bringing this back up. I was thinking about doing this myself, but I was going to hold off until the current round of PRs had been dealt with. But _somebody_ keeps opening helpful PRs. 😆

Speaking of holding off, I do think we should hold onto this one until those other PRs clear. I'd like to merge it in and then immediately tackle getting the codebase up to the linting standards. That way we avoid merge conflicts due to the formatting changes that the linter will suggest.

**Comment by dignite** [comment]
I suggest we merge it ASAP and then fix a few rules per week (could start with all the small ones), so that PRs can fix merge conflicts with ease. You never know who is building the next cool feature in a fork (not yet a PR) and fixing a bunch of rules at once could spoil that. Just my 2 cents :)

**Comment by dignite** [comment]
Yes we could not expect anyone to follow the rules until the linting is green, which I don't think is a huge problem. We would play catch up but sooner or later we would get to 0 and then we can make sure to keep it at zero. I have had great success fixing a few rules at a time before, that way fixing merge conflicts is just merging against the same type of change everywhere.

This PR is open for edit from you so feel free to make adjustments as you see fit though :)

**Comment by dignite** [comment]
Please have a look, this PR is not eslint violation free. 💃

**Pull request merged**